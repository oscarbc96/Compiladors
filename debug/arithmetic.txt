FLEX: Detected mode CALC
BISON: Starting in mode CALC
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 0 -> 0
BISON: ASSIGN ID: x TYPE: INT VALUE: 0
FLEX: ID yytex: x -> x
FLEX: ASSIGN yytex: :=
FLEX: ID yytex: x -> x
FLEX: ADD yytex: +
BISON: Looking for ID x
FLEX: INT yytex: 1 -> 1
BISON: Performing add
BISON: ASSIGN ID: x TYPE: INT VALUE: 1
FLEX: ASSIGN yytex: :=
FLEX: FLOAT yytex: 2.0 -> 2.000000
BISON: ASSIGN ID: a TYPE: FLOAT VALUE: 2.000000
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 5 -> 5
BISON: ASSIGN ID: b TYPE: INT VALUE: 5
FLEX: ASSIGN yytex: :=
FLEX: ID yytex: a -> a
FLEX: ADD yytex: +
BISON: Looking for ID a
FLEX: ID yytex: b -> b
BISON: Looking for ID b
BISON: Performing add
BISON: ASSIGN ID: c TYPE: FLOAT VALUE: 7.000000
FLEX: ID yytex: c -> c
FLEX: SUBSTRACT yytex: -
BISON: Looking for ID c
FLEX: ID yytex: a -> a
BISON: Looking for ID a
BISON: Performing substract
BISON: EXPRESSION: TYPE: FLOAT VALUE: 5.000000
FLEX: ASSIGN yytex: :=
FLEX: ID yytex: c -> c
FLEX: MULTIPLY yytex: *
BISON: Looking for ID c
FLEX: INT yytex: 3 -> 3
BISON: Performing multiply
BISON: ASSIGN ID: d TYPE: FLOAT VALUE: 21.000000
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: ID yytex: a -> a
BISON: Looking for ID a
FLEX: ADD yytex: +
FLEX: ID yytex: b -> b
BISON: Looking for ID b
FLEX: PARENTHESIS_CLOSE yytex: )
BISON: Performing add
FLEX: MULTIPLY yytex: *
FLEX: INT yytex: 3 -> 3
BISON: Performing multiply
BISON: EXPRESSION: TYPE: FLOAT VALUE: 21.000000
FLEX: INT yytex: 9 -> 9
FLEX: DIVIDE yytex: /
FLEX: INT yytex: 3 -> 3
BISON: Performing divide
BISON: EXPRESSION: TYPE: INT VALUE: 3
FLEX: ID yytex: a -> a
FLEX: POW yytex: **
BISON: Looking for ID a
FLEX: INT yytex: 3 -> 3
BISON: Performing pow
BISON: EXPRESSION: TYPE: FLOAT VALUE: 8.000000
FLEX: ID yytex: b -> b
FLEX: MOD yytex: mod
BISON: Looking for ID b
FLEX: INT yytex: 3 -> 3
BISON: Performing mod
BISON: EXPRESSION: TYPE: INT VALUE: 2
FLEX: STRING yytex: test -> test
FLEX: ADD yytex: +
FLEX: STRING yytex: - -> -
FLEX: ADD yytex: +
BISON: Performing add
FLEX: STRING yytex: test2 -> test2
BISON: Performing add
BISON: EXPRESSION: TYPE: STRING VALUE: test-test2
FLEX: STRING yytex: 4 + 4 =  -> 4 + 4 = 
FLEX: ADD yytex: +
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: INT yytex: 4 -> 4
FLEX: ADD yytex: +
FLEX: INT yytex: 4 -> 4
FLEX: PARENTHESIS_CLOSE yytex: )
BISON: Performing add
BISON: Performing add
BISON: EXPRESSION: TYPE: STRING VALUE: 4 + 4 = 8
The compilation has been successful
