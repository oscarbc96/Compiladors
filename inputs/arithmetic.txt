a := 2.0
b := 5

c := a + b

c - a

d := c * 3

(a + b) * 3

9 / 3

a ** 3

b mod 3

"test" + "-" + "test2"

"4 + 4 = " + (4 + 4)
