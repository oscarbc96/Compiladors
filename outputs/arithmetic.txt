ID: a TYPE: FLOAT VALUE: 2.000000
ID: b TYPE: INT VALUE: 5
ID: c TYPE: FLOAT VALUE: 7.000000
EXPRESSION: TYPE: FLOAT VALUE: -3.000000
EXPRESSION: TYPE: FLOAT VALUE: 5.000000
ID: d TYPE: FLOAT VALUE: 21.000000
EXPRESSION: TYPE: FLOAT VALUE: 21.000000
EXPRESSION: TYPE: INT VALUE: 3
EXPRESSION: TYPE: FLOAT VALUE: 8.000000
EXPRESSION: TYPE: INT VALUE: 2
EXPRESSION: TYPE: STRING VALUE: test-test2
EXPRESSION: TYPE: STRING VALUE: 4 + 4 = 8
ID: x TYPE: INT VALUE: -9
ID: x TYPE: INT VALUE: 9
EXPRESSION: TYPE: INT VALUE: -16
