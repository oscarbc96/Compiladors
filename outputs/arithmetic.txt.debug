FLEX: Detected mode CALC
BISON: Starting in mode CALC
FLEX: NEWLINE
FLEX: NEWLINE
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 0 -> 0
FLEX: NEWLINE
BISON: ASSIGN ID: x TYPE: INT VALUE: 0
FLEX: ID yytex: x -> x
FLEX: ASSIGN yytex: :=
FLEX: ID yytex: x -> x
FLEX: ADD yytex: +
BISON: Looking for ID x
FLEX: INT yytex: 1 -> 1
FLEX: NEWLINE
BISON: Performing add
BISON: ASSIGN ID: x TYPE: INT VALUE: 1
FLEX: NEWLINE
FLEX: ASSIGN yytex: :=
FLEX: FLOAT yytex: 2.0 -> 2.000000
FLEX: NEWLINE
BISON: ASSIGN ID: a TYPE: FLOAT VALUE: 2.000000
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 5 -> 5
FLEX: NEWLINE
BISON: ASSIGN ID: b TYPE: INT VALUE: 5
FLEX: NEWLINE
FLEX: ASSIGN yytex: :=
FLEX: ID yytex: a -> a
FLEX: ADD yytex: +
BISON: Looking for ID a
FLEX: ID yytex: b -> b
BISON: Looking for ID b
FLEX: NEWLINE
BISON: Performing add
BISON: ASSIGN ID: c TYPE: FLOAT VALUE: 7.000000
FLEX: NEWLINE
FLEX: ID yytex: c -> c
FLEX: SUBSTRACT yytex: -
BISON: Looking for ID c
FLEX: ID yytex: a -> a
BISON: Looking for ID a
FLEX: NEWLINE
BISON: Performing substract
BISON: EXPRESSION: TYPE: FLOAT VALUE: 5.000000
FLEX: NEWLINE
FLEX: ASSIGN yytex: :=
FLEX: ID yytex: c -> c
FLEX: MULTIPLY yytex: *
BISON: Looking for ID c
FLEX: INT yytex: 3 -> 3
FLEX: NEWLINE
BISON: Performing multiply
BISON: ASSIGN ID: d TYPE: FLOAT VALUE: 21.000000
FLEX: NEWLINE
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: ID yytex: a -> a
BISON: Looking for ID a
FLEX: ADD yytex: +
FLEX: ID yytex: b -> b
BISON: Looking for ID b
FLEX: PARENTHESIS_CLOSE yytex: )
BISON: Performing add
FLEX: MULTIPLY yytex: *
FLEX: INT yytex: 3 -> 3
FLEX: NEWLINE
BISON: Performing multiply
BISON: EXPRESSION: TYPE: FLOAT VALUE: 21.000000
FLEX: NEWLINE
FLEX: INT yytex: 9 -> 9
FLEX: DIVIDE yytex: /
FLEX: INT yytex: 3 -> 3
FLEX: NEWLINE
BISON: Performing divide
BISON: EXPRESSION: TYPE: INT VALUE: 3
FLEX: NEWLINE
FLEX: ID yytex: a -> a
FLEX: POW yytex: **
BISON: Looking for ID a
FLEX: INT yytex: 3 -> 3
BISON: Performing pow
FLEX: NEWLINE
BISON: EXPRESSION: TYPE: FLOAT VALUE: 8.000000
FLEX: NEWLINE
FLEX: ID yytex: b -> b
FLEX: MOD yytex: mod
BISON: Looking for ID b
FLEX: INT yytex: 3 -> 3
FLEX: NEWLINE
BISON: Performing mod
BISON: EXPRESSION: TYPE: INT VALUE: 125
FLEX: NEWLINE
FLEX: STRING yytex: test -> test
FLEX: ADD yytex: +
FLEX: STRING yytex: - -> -
FLEX: ADD yytex: +
BISON: Performing add
FLEX: STRING yytex: test2 -> test2
FLEX: NEWLINE
BISON: Performing add
BISON: EXPRESSION: TYPE: STRING VALUE: test-test2
FLEX: NEWLINE
FLEX: STRING yytex: 4 + 4 =  -> 4 + 4 = 
FLEX: ADD yytex: +
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: INT yytex: 4 -> 4
FLEX: ADD yytex: +
FLEX: INT yytex: 4 -> 4
FLEX: PARENTHESIS_CLOSE yytex: )
BISON: Performing add
FLEX: NEWLINE
BISON: Performing add
BISON: EXPRESSION: TYPE: STRING VALUE: 4 + 4 = 8
The compilation has been successful
