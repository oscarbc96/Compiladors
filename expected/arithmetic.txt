ASSIGN ID: x TYPE: INT VALUE: 0
ASSIGN ID: x TYPE: INT VALUE: 1
ASSIGN ID: a TYPE: FLOAT VALUE: 2.000000
ASSIGN ID: b TYPE: INT VALUE: 5
ASSIGN ID: c TYPE: FLOAT VALUE: 7.000000
EXPRESSION: TYPE: FLOAT VALUE: 5.000000
ASSIGN ID: d TYPE: FLOAT VALUE: 21.000000
EXPRESSION: TYPE: FLOAT VALUE: 21.000000
EXPRESSION: TYPE: INT VALUE: 3
EXPRESSION: TYPE: FLOAT VALUE: 8.000000
EXPRESSION: TYPE: INT VALUE: 2
EXPRESSION: TYPE: STRING VALUE: test-test2
EXPRESSION: TYPE: STRING VALUE: 4 + 4 = 8
